HELM-326: Add documentation for location() template_variable (#415)
OpenNMS/opennms-helm: docs/modules/datasources/pages/flow_datasource.adoc

[[ds-flow-template]]
== Template queries

=== locations()

The `locations()` query returns a list of all available node locations. 

> Use this variable with exporterNodesWithFlows to reduce the number of nodes displayed by location.

Reviewer: ```suggestion
Use this variable with `exporterNodesWithFlows()` to reduce the number of nodes displayed by location.
```

> Use this variable with exporterNodesWithFlows to reduce the number of nodes displayed by location.

Reviewer 2: ```suggestion
Use this variable with `exporterNodesWithFlows` to reduce the number of nodes displayed by location.
```

> Use this variable with exporterNodesWithFlows to reduce the number of nodes displayed by location.

Reviewer 2: ```suggestion
Use this variable with `exporterNodesWithFlows` to reduce the number of nodes displayed by location.
```

> Examples: +

Reviewer 2: ```suggestion
Examples:
```

> Examples: +
> ```
> exporterNodesWithFlows(location='Default')
> exporterNodesWithFlows(location='$location')
> ```

Reviewer: ```suggestion
Examples:

* exporterNodesWithFlows(location='Default')
* exporterNodesWithFlows(location='$location')
```

=== exporterNodesWithFlows(<filterQuery>)

> The `exporterNodesWithFlows()` query will return a list of nodes which have exported flow data to OpenNMS.

Reviewer 2: ```suggestion
The `exporterNodesWithFlows()` query returns a list of nodes which have exported flow data to OpenNMS.
```

[[ds-flow-template]]
== Template queries

=== locations()

The `locations()` query returns a list of all available node locations. 

> Use this variable with `exporterNodesWithFlows()` to reduce the number of nodes displayed by location.

Reviewer: ```suggestion
Use this inside `exporterNodesWithFlows()` to filter the number of nodes displayed by location.
```

> Examples:
> ```
> exporterNodesWithFlows(location='Default')
> exporterNodesWithFlows(location='$location')
> ```

Reviewer: ```suggestion
.Example variable expression
* exporterNodesWithFlows(location='Default')
* exporterNodesWithFlows(location='$location')
```

exporterNodesWithFlows()
```

> The filter expression must be an existing node attribute and their corresponding value and the value can be constant or a template_variable.

Reviewer 2: ```suggestion
The filter expression must be an existing node attribute and its corresponding value.
The value can be a constant or a template_variable.
```

> Example: +

Reviewer 2: ```suggestion
Example:
```

> Example: +
> ```
> exporterNodesWithFlows(location='Default')
> ```

Reviewer: ```suggestion
Example:

* exporterNodesWithFlows(location='Default')
```

=== interfacesOnExporterNodeWithFlows(<nodeCriteria>)

> The `interfacesOnExporterNodeWithFlows()` query will return a list of interface indexes which are associated with one or more flows on the given node.

Reviewer 2: ```suggestion
The `interfacesOnExporterNodeWithFlows()` query returns a list of interface indexes which are associated with one or more flows on the given node.
```

exporterNodesWithFlows()
```

The filter expression must be an existing node attribute and its corresponding value.

> The value can be a constant or a template_variable.

Reviewer: ```suggestion
The value can be a constant or a variable.
```

> Example:
> ```
> exporterNodesWithFlows(location='Default')
> ```

Reviewer: ```suggestion
.Example variable expression
exporterNodesWithFlows(location='Default')
```
